### PR TITLE
support x86 (32bits) arch

### PR DESCRIPTION
**Pull request by jun-zeng**
Please use read_trace_zj.py in pypeekaboo folder to read x86 (32 bits) trace. I have generated a 'ls' trace in Ubuntu 16.04 32bits VM for you to test the results. You can download it with the following link. 

https://drive.google.com/open?id=1coiILZ_6BjBqr6Ij01oIAG_ZHalWQM39
MD5 checksum: 949e473efa552ff76a0d1901ac7fd7be

### CODE REVIEW

**Review by mimicji — CHANGES_REQUESTED** [review]
Please clean up all temp files before merging. 

Thanks.

**Review by melynx — COMMENTED** [review]
Why do you need to specifically create a 32bit store for the buffer?

**Comment by jun-zeng** [comment]
This is because the size of pc in X86 arch is 4 bytes. If the program still uses OPND_CREATE_INT64 in drx_buf_insert_buf_store function, it will cause incompatible type error. 

We can also use type casting to solve this incompatibility problem if you don't like 32-bit buffer.

compile_error

**Review by melynx — CHANGES_REQUESTED** [review]
dynamorio defines X64 for 64bit architectures (both ARM and X86). The #ifdef should be using X64 and you probably only need to #ifdef the buf_store.

Also, merge the X86 support into pypeekaboo and use the arch from metafile to determine how to parse it.

**Comment by melynx** [comment]
What about the C structs for x86?

**Comment by jun-zeng** [comment]
It's located in arch folder

File: peekaboo_dr/peekaboo_dr.c [hunk 1]
```diff
@@ -435,6 +453,6 @@ DR_EXPORT void dr_client_main(client_id_t id, int argc, const char *argv[])
 	printf("Peekaboo: %s - Client 'peekaboo' initializing\n", arch_str);
 
 	printf("Peekaboo: Binary being traced: %s\n", dr_get_application_name());
-	printf("Peekaboo: Number of SIMD slots: %d\n", MCXT_NUM_SIMD_SLOTS);
+	// printf("Peekaboo: Number of SIMD slots: %d\n", MCXT_NUM_SIMD_SLOTS);
```
Comment (melynx) [inline peekaboo_dr/peekaboo_dr.c]: Why is this commented. Uncomment this!

**Review by melynx — CHANGES_REQUESTED** [review]

**Review by mimicji — APPROVED** [review]
Accept the pull request.
Thanks, Jun!

**Pull request merged**